cupy/core/_routines_math.pyx

```diff
@@ -519,9 +519,10 @@ cpdef scan_core(ndarray a, axis, scan_op op, dtype=None, ndarray out=None):
         result[...] = a
```

Comment:
It was not part of the changes in this PR, but can't this else clause from lines 515-518 look like the following instead?


```Python
        if (out.flags.c_contiguous or out.flags.f_contiguous):
            result = out
            result[...] = a
        else:
            result = a.astype(out.dtype, copy=False)
```

Reply:
Agreed. It looks to me too that `result[...] = a` is not necessary for the `else` path.

Reply:
However, I don't think it is a good idea to add `copy=False` to the `a.astype` in the `else` path above, as then the input array `a` might be overwritten in the following `scan`.
https://github.com/cupy/cupy/blob/517081f9ad80b5036879572cc03137d25c1da710/cupy/core/_routines_math.pyx#L538